<?xml version="1.0" ?>
<launch>
	<arg name="rgb_topic"         default="/camera/color/image_raw"/>
	<arg name="depth_topic"       default="/camera/aligned_depth_to_color/image_raw"/>
	<arg name="camera_info_topic" default="/camera/color/camera_info"/>
	<arg name="approx_sync"       default="false"/>

	<group ns="rtabmap">
		<node name="rtabmap" pkg="rtabmap_ros" type="rtabmap" output="screen" args="--delete_db_on_start">
			<param name="frame_id"      type="string" value="panda_link0"/> <!--base frame of the arm-->
			<param name="odom_frame_id" type="string" value="panda_link0"/> <!-- same as frame_id, as the base of the arm doesn t move -->
			<param name="approx_sync"   type="bool"   value="$(arg approx_sync)"/>

			<remap from="rgb/image"       to="$(arg rgb_topic)"/>
			<remap from="depth/image"     to="$(arg depth_topic)"/>
			<remap from="rgb/camera_info" to="$(arg camera_info_topic)"/>

			<!-- RTAB-Map parameters -->
			<param name="Rtabmap/DetectionRate"     type="double" value="1"/>     <!-- Hz -->
			<param name="Kp/MaxFeatures"            type="double" value="-1"/>    <!-- Disable loop closure detection -->
			<param name="RGBD/ProximityBySpace"     type="bool" value="false"/> <!-- Disable proximity detection -->
			<param name="RGBD/AngularUpdate"        type="double" value="0.0"/>   <!-- Always add frames -->
			<param name="RGBD/LinearUpdate"         type="double" value="0.0"/>   <!-- Always add frames -->
			<param name="Grid/CellSize"				type="double" value="0.005"/>
			<param name="Grid/NormalsSegmentation"	type="bool"	value="false"/>
			<param name="Grid/MaxObstacleHeight"	type="double" value="0.2"/>
			<param name="Grid/MaxGroundHeight"		type="double" value="0.035"/>
			<param name="Grid/NoiseFilteringRadius"	type="double" value="0.01"/>
			<param name="Grid/RangeMax"				type="double" value="0.4"/>
			
			<!-- Map parameters -->
			<param name="~map_always_update"	type="bool" value="true"/>
			<param name="~cloud_output_voxelized"	type="bool" value="true"/>
			<!--<param name="Grid/VoxelSize"           type="string" value="0.001"/>-->	<!-- Voxelization of the cloud_map -->

		</node>
	</group>
    
    <node pkg="robot_arm_3Dscan" type="RTABMAPServer" name="rtabmap_server" output="screen"/>
</launch>